src/TypeScript.UnitTests/NodeJSLocator/LocationProviders/BundledNodeLocationsProviderTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Runtime.InteropServices;
+using FluentAssertions;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator.LocationProviders;
+
+namespace SonarLint.VisualStudio.TypeScript.UnitTests.NodeJSLocator.LocationProviders
+{
+    [TestClass]
+    public class BundledNodeLocationsProviderTests
+    {
+        [TestMethod]
+        public void Get_FailsToRetrieveInstallDirectory_Exception()
+        {
+            const string installDir = "c:\\test\\";
+
+            var testSubject = CreateTestSubject(installDir, shellHrResult: VSConstants.E_FAIL);
+            Action act = () => testSubject.Get();
+
+            act.Should().Throw<COMException>();
```

Comment:
Where do we catch this? We don't want a gold bar to appear complaining SLVS.
It would be simpler to just return an empty list in this case.